File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.js
function solution(sizes) {    
    const result = [0,0]; 

    sizes.forEach(inner => {
        inner.sort((a,b)=>a-b)
        inner.forEach((value, idx) => {
            if(result[idx] < value) result[idx] = value;
  });
});
    
    return result.reduce((acc,cur) => acc*cur);
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    # 가로를 최대, 세로를 최솟값으로 정렬 수 가로, 세로 결정
    max_width, max_height = 0,0
    # print(max_width,max_height)
    
    for w,h in sizes:
        # print(w, h)
        
        if h > w :
            w, h = h,w
        
        if max_width < w:
            max_width = w
        
        if max_height < h:
            max_height = h
    
    return max_width * max_height